waspc/data/Generator/templates/react-app/src/operations/resources.js

```diff
@@ -45,3 +45,22 @@ export const invalidateQueriesUsing = (resources) => {
     queryClient.invalidateQueries(queryCacheKey)
   }
 }
+
+export function removeQueries() {
+  // TODO(filip): We are currently removing all the queries, but we should
+  // remove only non-public, user-dependent queries - public queries are
+  // expected not to change in respect to the currently logged in user.
+  queryClient.removeQueries()
+}
+
+// TODO(filip): We are currently invalidating and removing  all the queries, but
+// we should remove only the non-public, user-dependent ones.
+export function invalidateAndClearQueries() {
```

Comment:
So looking at this function vs the other, removeQueries function -> how would you explain to me what is the difference between them, when should I use each of them? How do I decide which one I want to use?

Also, if the first one only calls another function with the same name, and the second one merely calls two functions one after each other and is named exactly like those two concatenated, what is the purpose of that? Why don't I just call the queryClient methods directly?

Reply:
> Also, if the first one only calls another function with the same name, and the second one merely calls two functions one after each other and is named exactly like those two concatenated, what is the purpose of that? Why don't I just call the queryClient methods directly?

The only purpose is to minimize coupling with the `queryClient`. Since the `queryClient` is going to be set asynchronously in the next PR, its usage will become more complicated. It's better if we expose it to as few files as possible.

Reply:
Thats ok, I don't think leaking is the big issue here.
You could, however, just expose two functions that are "invalidate" and "remove" and then use them in the code independently. But ok, I don't have a good idea what is better nor do I grasp the whole situation with queryClient right now, so let's go with whatever you think make most sense!